### PR TITLE
[buffer] harmonize geographic strategy code and share code

### PR DESCRIPTION
**Pull request by barendgehrels**
As discussed in last PR

### CODE REVIEW

File: include/boost/geometry/strategies/geographic/buffer_join_round.hpp [hunk 1]
```diff
@@ -71,52 +59,34 @@ public :
...
+        if (! helper::calculate_angles(lon_rad, lat_rad, perp1, perp2, m_spheroid,
+                                       angle_diff, first_azimuth))
         {
-            // Defensive check with asserts
-            BOOST_GEOMETRY_ASSERT(angle_diff >= 0);
-            BOOST_GEOMETRY_ASSERT(angle_diff <= geometry::math::pi<calc_t>());
             return false;
         }
 
+        static calc_t const two_pi = geometry::math::two_pi<calc_t>();
```
Comment (awulkiew) [inline include/boost/geometry/strategies/geographic/buffer_join_round.hpp:75]: This is probably out of scope of this PR but we could replace `static` variables with `constexpr` variables. For this to work math constants would have to be `constexpr` functions.

File: include/boost/geometry/strategies/geographic/buffer_side_straight.hpp [hunk 2]
```diff
@@ -52,21 +54,23 @@ public :
     template
     <
         typename Point,
-        typename OutputRange,
+        typename RangeOut,
```
Comment (awulkiew) [inline include/boost/geometry/strategies/geographic/buffer_side_straight.hpp:57]: Is there a reason why template arguments are in different order than function arguments?

**Review by awulkiew — APPROVED** [review]

File: include/boost/geometry/strategies/geographic/buffer_point_circle.hpp [hunk 3]
```diff
@@ -55,77 +59,67 @@ template
...
+
     //! \brief Constructs the strategy
-    //! \param count number of points for the created circle (if count
-    //! is smaller than 3, count is internally set to 3)
-    explicit geographic_point_circle(std::size_t count = 90)
-        : m_count((count < 3u) ? 3u : count)
+    //! \param points_per_circle Number of points for a full circle
+    //! (if points_per_circle is smaller than 4, it is internally set to 4)
+    explicit geographic_point_circle(std::size_t points_per_circle = 90)
+        : m_points_per_circle((points_per_circle < 4u) ? 4u : points_per_circle)
```
Comment (vissarion) [inline include/boost/geometry/strategies/geographic/buffer_point_circle.hpp:67]: what is the point of changing this from 3 to 4? Is this for robustness?

**Review by vissarion — APPROVED** [review]

File: include/boost/geometry/strategies/geographic/buffer_join_round.hpp (hunk 1 above)
Comment (barendgehrels) [reply include/boost/geometry/strategies/geographic/buffer_join_round.hpp:75]: Indeed, I'll do that later in a separate PR

File: include/boost/geometry/strategies/geographic/buffer_side_straight.hpp (hunk 2 above)
Comment (barendgehrels) [reply include/boost/geometry/strategies/geographic/buffer_side_straight.hpp:57]: Thanks, no, I'll update it.

File: include/boost/geometry/strategies/geographic/buffer_point_circle.hpp (hunk 3 above)
Comment (barendgehrels) [reply include/boost/geometry/strategies/geographic/buffer_point_circle.hpp:67]: One had 3, the other had 4. I liked to harmonize it. I can make both `3` too. Either choice is fine, it's very low anyway, not really circle-like.

File: include/boost/geometry/strategies/geographic/buffer_side_straight.hpp (hunk 2 above)
Comment (barendgehrels) [reply include/boost/geometry/strategies/geographic/buffer_side_straight.hpp]: :heavy_check_mark:

File: include/boost/geometry/strategies/geographic/buffer_point_circle.hpp (hunk 3 above)
Comment (barendgehrels) [reply include/boost/geometry/strategies/geographic/buffer_point_circle.hpp]: For the join it was also 3 but that was erroneous, I made it minimum 4 to get well-shaped symmetric joins.
I don't think it was ever used with 3 (default = 90)

**Pull request merged**